crates/exex/src/exex.rs

```diff
@@ -185,6 +196,73 @@ impl<Node: FullNodeComponents> KakarotRollup<Node> {
     ) -> eyre::Result<()> {
         self.db.insert_execution_trace(number, trace, memory, air_public_input, air_private_input)
     }
+
+    /// Executes the Cairo program to retrieve and process the environment (block and transaction)
+    /// data.
+    ///
+    /// This function takes a sealed block and a list of signed transactions, executes a Cairo
+    /// program to retrieve environment data via a hint.
+    fn execute_get_env(
+        &mut self,
+        block: SealedBlock,
+        txs: Vec<TransactionSignedEcRecovered>,
+        chain_tip: u64,
+    ) -> eyre::Result<()> {
+        // Initialize the Cairo runtime configuration with various settings.
+        let config = CairoRunConfig {
+            layout: LayoutName::all_cairo,
+            trace_enabled: true,
+            relocate_mem: true,
+            proof_mode: true,
+            allow_missing_builtins: Some(false), // Ensures no missing built-in hints are allowed.
+            ..Default::default()
+        };
+
+        // Loop through each transaction in the block to run the Cairo program.
+        for tx in &txs {
```

Comment:
There will be one Cairo run for the whole block, not one per tx

Reply:
As long as we have to fill the `env` with the transaction fields we leave it like that right? Or do you prefer that I fill with 0 the sender, chain id and gas price so that we can execute this only once a block?

I did this as a first draft version to be able to complete the whole env `struct`